docs/spark-connector/spark-catalog-iceberg.md

```diff
@@ -74,18 +78,62 @@ USING (SELECT 4 as id, 'David' as name, 'Engineering' as department, TIMESTAMP '
-| `catalog-backend`               | `type`                                | Catalog backend type      | 0.5.0         |
-| `uri`                           | `uri`                                 | Catalog backend uri       | 0.5.0         |
-| `warehouse`                     | `warehouse`                           | Catalog backend warehouse | 0.5.0         |
-| `jdbc-user`                     | `jdbc.user`                           | JDBC user name            | 0.5.0         |
-| `jdbc-password`                 | `jdbc.password`                       | JDBC password             | 0.5.0         |
+#### HiveCatalog
+
+| Gravitino catalog property name | Spark Iceberg connector configuration | Default Value | Required | Description               | Since Version |
+|---------------------------------|---------------------------------------|---------------|----------|---------------------------|---------------|
+| `catalog-backend`               | `type`                                | `hive`        | Yes      | Catalog backend type      | 0.5.0         |
+| `uri`                           | `uri`                                 | (none)        | Yes      | Catalog backend uri       | 0.5.0         |
+| `warehouse`                     | `warehouse`                           | (none)        | Yes      | Catalog backend warehouse | 0.5.0         |
+
+#### JdbcCatalog
+
+| Gravitino catalog property name | Spark Iceberg connector configuration | Default Value | Required | Description               | Since Version |
+|---------------------------------|---------------------------------------|---------------|----------|---------------------------|---------------|
+| `catalog-backend`               | `type`                                | `jdbc`        | Yes      | Catalog backend type      | 0.5.0         |
+| `uri`                           | `uri`                                 | (none)        | Yes      | Catalog backend uri       | 0.5.0         |
+| `warehouse`                     | `warehouse`                           | (none)        | Yes      | Catalog backend warehouse | 0.5.0         |
+| `jdbc-user`                     | `jdbc.user`                           | (none)        | Yes      | JDBC user name            | 0.5.0         |
+| `jdbc-password`                 | `jdbc.password`                       | (none)        | Yes      | JDBC password             | 0.5.0         |
+
+#### RESTCatalog
+
+| Gravitino catalog property name | Spark Iceberg connector configuration | Default Value | Required | Description               | Since Version |
+|---------------------------------|---------------------------------------|---------------|----------|---------------------------|---------------|
+| `catalog-backend`               | `type`                                | `rest`        | Yes      | Catalog backend type      | 0.5.1         |
```

Comment:
RESTCatalog is supported in `0.5.1` while `catalog-backend` is supported in `0.5.0`, so what's the value for `Since Version` of `catalog-backend` for RESTCatalog? @jerryshao

Reply:
change the value of `Since Version` to `0.5.0` and add an extra description to indicate that `RESTCatalog` is supported since `0.5.1`. WDYT? @FANNG1

Reply:
I think it should be 0.5.0, the parameter is introduced in 0.5.0, whereas we extend to support rest in 0.5.1, so only update the value should be enough.

Reply:
On a secondary check, I think 0.5.1 makes more sense, so I'm OK with the current changes.